ratis-client/src/main/java/org/apache/ratis/client/impl/DataStreamClientImpl.java

```diff
@@ -105,12 +107,13 @@ private DataStreamOutputImpl(RaftClientRequest request) {
       this.header = request;
       final ByteBuffer buffer = ClientProtoUtils.toRaftClientRequestProtoByteBuffer(header);
       this.headerFuture = send(Type.STREAM_HEADER, buffer, buffer.remaining());
+      this.slidingWindow = new SlidingWindow.Client<>(ClientInvocationId.valueOf(clientId, header.getCallId()));
```

Comment:
Oops, we need to initialize the slidingWindow before sending the header, i.e.
```
      this.header = request;
      this.slidingWindow = new SlidingWindow.Client<>(ClientInvocationId.valueOf(clientId, header.getCallId()));
      final ByteBuffer buffer = ClientProtoUtils.toRaftClientRequestProtoByteBuffer(header);
      this.headerFuture = send(Type.STREAM_HEADER, buffer, buffer.remaining());
```

Reply:
Has been modified @szetszwo @captainzmc